### DISCUSSION TITLE
Account feature: Sign message

### DISCUSSION THREAD

**Discussion by taurenshaman** [idena-network/idena-docs#109] [IIPs]
For a normal wallet of any blockchain, `Sign Message` is a very important feature.
Advantage: 
* A customary way to prove the ownership of the behavior/action of Signing some specific Message
* No gas
* Not commit to blockchain
* Replaces Account/Password mode
* A necessary way for DApps

Share of some common mechanisms:
### Ethereum / EVM
* message = some text
* signature = hash(message)

**usage**: get the owner address
address = ec_recover(message, signature)

### Solona
* message = some text
* publicKey = SOL Address
* signature is JSON: 
```
{"signature":{"type":"Buffer","data":[ // Uint8Array
]},
"publicKey":""}
```
**usage**: verify message and signature by publicKey

### Arweave
* message = some text
* signature is Uint8Array
* publicKey = JWK.n
![image](https://user-images.githubusercontent.com/1308706/199238276-76150f02-193a-4322-ae40-49982c109e92.png)

**usage**: verify message and signature by publicKey

### Snapshots
![image](https://user-images.githubusercontent.com/1308706/199239549-c4c02b39-7690-41c3-9c22-aee2d112d34b.png)
![image](https://user-images.githubusercontent.com/1308706/199239912-ca0569a3-e049-4b65-9067-3a2481360636.png)

**Comment by midenaio** [comment]
@taurenshaman How urgently is this feature needed? Do you need the message signing function for your app development or is it a general feature request for some future needs?

**Reply by taurenshaman** [reply]
Yes, my DApp needs that feature. I'm developing a mutable DID project: [DNA](https://dna.runebox.xyz/), which makes another kind of muti-sign: ETH wallet + Arweave wallet + Solana wallet (+Idena, if it supports signing message)
Now I can only connect an Idena account, but cannot use Idena to sign a message.

![22b7199d42e134415481a9c7520a038](https://user-images.githubusercontent.com/1308706/200171325-2d5ce34c-7fa4-4c42-8e0d-6f7d60a15270.png)
